src/redprl/refiner_types.fun

```diff
@@ -1268,6 +1270,176 @@ struct
       end
   end
 
+  structure FormalComposition =
+  struct
+    val kindConstraintOnCapAndTubes =
+      fn K.DISCRETE => (K.DISCRETE, K.DISCRETE)
+       | K.KAN => (K.KAN, K.KAN) (* XXX more research needed *)
```

Comment:
Does `more research needed` mean that these conditions are sufficient but maybe not necessary? Or the other way around?

Reply:
I hope it's the former, but in some unfortunate cases it could be the latter... the fcom types are so complex that I am not sure about anything. `K.KAN => (K.KAN, K.KAN)` should be correct though.

Reply:
OK! We can improve and correct if needed in the future as we begin to understand these things in a more precise way.